### PR TITLE
Reusable modules for EC2, RDS and Network

### PR DESCRIPTION
**Pull request by Debashbora**
Added Example folder to depict usage of EC2, RDS, and Network modules created earlier:
Screenshot from 2024-05-27 14-43-10
Screenshot from 2024-05-27 14-42-37
Screenshot from 2024-05-27 14-42-28
Screenshot from 2024-05-27 14-42-20

### CODE REVIEW

File: terraform/example/ec2/data.tf [hunk 1]
```diff
@@ -0,0 +1,13 @@
+data "aws_vpc" "poc_vpc" {
```
Comment (mayank0202) [inline terraform/example/ec2/data.tf:1]: fetch ami from data lookups

File: terraform/example/rds/main.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  egress_rules      = var.egress_rules
+  #   backup_retention_period     = var.backup_retention_period
+  multi_az = var.multi_az
+  #   delete_automated_backups    = var.delete_automated_backups
+  #   publicly_accessible         = var.publicly_accessible
+  skip_final_snapshot = var.skip_final_snapshot
+  #   apply_immediately           = var.apply_immediately
+  #   environment                 = var.environment
+  #   namespace                   = var.namespace
+}
```
Comment (mayank0202) [inline terraform/example/rds/main.tf]: add tags also

**Review by mayank0202 — CHANGES_REQUESTED** [review]
run pre-commit please

File: terraform/example/rds/main.tf (hunk 2 above)
Comment (Debashbora) [reply terraform/example/rds/main.tf]: done

File: terraform/example/ec2/data.tf (hunk 1 above)
Comment (Debashbora) [reply terraform/example/ec2/data.tf:1]: done, Kindly check

**Review by mayank0202 — COMMENTED** [review]
LGTM

**Review by mayank0202 — APPROVED** [review]
LGTM

**Pull request merged**